src/Microsoft.AspNetCore.Authentication/Internal/EventSpan.cs

```diff
@@ -0,0 +1,83 @@
+using System;
+using System.Diagnostics;
+
+namespace Microsoft.AspNetCore.Authentication.Internal
+{
+    // REVIEW: This file will be in a common shared-source package.
+    internal struct ValueStopwatch
+    {
+        private static readonly double TimestampToTicks = TimeSpan.TicksPerSecond / (double)Stopwatch.Frequency;
+
+        private bool _enabled;
```

Comment:
Do you really need `_enabled`?

Reply:
Probably not, now that I just decided to use `_startTimestamp == 0` to indicate uninitialized. I could just do the same with `_endTimestamp == 0` implying Enabled.

Reply:
Unlike `Stopwatch`, `ValueStopwatch` is single-use and once it's stopped it can't be resumed or reset (both of which `Stopwatch` does allow)